tests/test_env.py

```diff
@@ -68,11 +72,10 @@ def test_env_accessors(gdalenv):
 
 
 def test_ensure_env_decorator(gdalenv):
+    @ensure_env
     def f():
         return getenv()['WITH_RASTERIO_ENV']
-    wrapper = ensure_env(f)
-    assert wrapper() is True
-    assert 'WITH_RASTERIO_ENV' not in getenv()
```

Comment:
This test relied on the buggy behavior this PR fixes.  After the function executes the environment should be empty.